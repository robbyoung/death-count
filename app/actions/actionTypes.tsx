export enum ActionType {
    ADD_PLAYTHROUGH = 'Add Playthrough',
    ADD_GAME = 'Add Game',
    ADD_DEATH = 'Add Death',
    ADD_DEATH_DETAIL = 'Add Death Detail',
    COMPLETE_DEATH = 'Complete Death',
    ADD_OPTION = 'Add Option',
    ADD_OPTION_SET = 'Add Option Set',
    SELECT_GAME = 'Select Game',
    LOAD_STATE = 'Load State',
    COMPLETE_GAME = 'Complete Game',
    SELECT_PLAYTHROUGH = 'Select Playthrough',
    SELECT_OPTION_SET = 'Select Option Set',
    DELETE_PLAYTHROUGH = 'Delete Playthrough',
}
