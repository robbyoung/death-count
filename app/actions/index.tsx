export * from './actionTypes';
export * from './addPlaythrough';
export * from './addDeath';
export * from './addGame';
export * from './addDeathDetail';
export * from './completeDeath';