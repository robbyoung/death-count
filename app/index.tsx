import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Screens } from './screens';
import Home from './containers/home';
import NewDeath from './containers/newDeath';
import GamesList from './containers/gamesList';
import NewGame from './containers/newGame';
import Loading from './containers/loading';
import Startup from './containers/startup';
import PlaythroughsList from './containers/playthroughsList';

const MainNavigator = createStackNavigator(
	{
		Loading: { screen: Loading },
		Home: { screen: Home },
		NewDeath: { screen: NewDeath },
		GamesList: { screen: GamesList },
		NewGame: { screen: NewGame },
		Startup: { screen: Startup },
		PlaythroughsList: { screen: PlaythroughsList }
	},
	{
		initialRouteName: Screens.Loading,
	},
);

export default createAppContainer(MainNavigator);