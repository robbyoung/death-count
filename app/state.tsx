export interface Game {
    id: string;
    name: string;
    options: string;
}

export interface DeathOptions {
    name: string;
    options: string[];
}

export interface Playthrough {
    id: string;
    name: string;
    description: string;
    gameId: string;
}

export interface Death {
    id: string;
    playthroughId: string;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    currentPlaythrough: string;
    currentGame: string;
}