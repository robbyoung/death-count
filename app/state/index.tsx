export * from './state';
export * from './testState';