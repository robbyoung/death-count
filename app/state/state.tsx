export interface Game {
    id: string;
    name: string;
    selected: boolean;
    complete: boolean;
}

export interface Playthrough {
    id: string;
    name: string;
    gameId: string;
    selected: boolean;
}

export interface Death {
    id: string;
    playthroughId: string;
    complete: boolean;
    details: { [id: string] : string };
}

export interface OptionSet {
    id: string;
    gameId: string;
    title: string;
    options: string[];
    selected: boolean;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    optionSets: OptionSet[];
}