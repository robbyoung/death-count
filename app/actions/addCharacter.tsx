import { Death, Character } from '../state';
import { Action } from 'redux';

export interface AddCharacterAction extends Action {
    newCharacter: Character;
}

export function addCharacter(state: Character[], action: AddCharacterAction): Character[] {
    return [
        ...state,
        action.newCharacter,
    ]
}