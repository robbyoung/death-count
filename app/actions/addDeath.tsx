import { Death } from '../state';
import { Action } from 'redux';
import { ActionType } from './actionTypes';

export class AddDeath implements Action {
    type: ActionType = ActionType.ADD_DEATH;

    constructor(public newDeath: Death) {}
}

export function addDeath(state: Death[], action: AddDeath): Death[] {
    return [
        ...state,
        action.newDeath,
    ]
}