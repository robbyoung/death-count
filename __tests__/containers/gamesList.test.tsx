import React from 'react';
import renderer from 'react-test-renderer';
import GamesList from '../../app/containers/gamesList';
import store from '../../app/store';
import { loadStateAction } from '../../app/actions';
import { createTestState } from '../../app/state';

const fakeNavigation = {
    goBack: () => undefined,
}
jest.mock('@react-native-community/async-storage', () => ({}));

describe('Games List Container', () => {
    it('Renders with the empty initial state', () => {
        const component = renderer.create(<GamesList navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Renders with games in the state', () => {
        const state = createTestState(1, 1, 2, 0, 0);
        const action = loadStateAction(state);
        store.dispatch(action);
        
        const component = renderer.create(<GamesList navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
