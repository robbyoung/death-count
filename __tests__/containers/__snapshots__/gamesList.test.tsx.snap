// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games List Container Renders with games in the state 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#e84641",
            "borderBottomColor": "#592a29",
            "borderBottomWidth": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 22,
              "margin": 10,
            }
          }
        >
          Game 0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Add a new game"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "flexGrow": 100,
            "fontSize": 22,
            "height": 50,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": undefined,
              },
              Object {
                "backgroundColor": "#e84641",
                "borderBottomRightRadius": 10,
                "borderTopRightRadius": 10,
                "color": "#ffffff",
                "fontSize": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              },
              Object {
                "fontFamily": "FontAwesome5FreeSolid",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Games List Container Renders with the empty initial state 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignContent": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Add a new game"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "flexGrow": 100,
            "fontSize": 22,
            "height": 50,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": undefined,
              },
              Object {
                "backgroundColor": "#e84641",
                "borderBottomRightRadius": 10,
                "borderTopRightRadius": 10,
                "color": "#ffffff",
                "fontSize": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              },
              Object {
                "fontFamily": "FontAwesome5FreeSolid",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
